src/main/java/NyanBot.java

```diff
@@ -0,0 +1,237 @@
+                case MARK_COMMAND:
+                    int markIndex;
+                    markTask(input);
+                    writeFile();
+                    break;
+                case UNMARK_COMMAND:
+                    unmarkTask(input);
+                    writeFile();
+                    break;
+                case TODO_COMMAND:
+                    addTodo(input);
+                    writeFile();
+                    break;
+                case DEADLINE_COMMAND:
+                    addDeadline(input);
+                    writeFile();
+                    break;
+                case EVENT_COMMAND:
+                    addEvent(input);
+                    writeFile();
+                    break;
+                case HELP_COMMAND:
+                    Printer.printSike();
+                    break;
+                default:
+                    Printer.printInvalidInput();
+                    break;
+            }
+        }
+    }
+
+    private static void readFile() {
+        try {
+            List<String> lines = FileHandler.readFile("data/nyan.txt");
+            for (String line : lines) {
+                tasks.add(readLine(line));
+            }
+        } catch (IOException e) {
+            System.out.println("IO Exception");
+        }
+    }
+
+    private static Task readLine(String line) {
+        String[] tokens = line.split("/");
+        String command = tokens[0].toUpperCase();
+        String status = tokens[1].toUpperCase();
+        Task task = new Task("");
+        switch (command) {
+            case TODO_COMMAND:
+                task = new Todo(tokens[2]);
+                break;
+            case DEADLINE_COMMAND:
+                task = new Deadline(tokens[2], tokens[3]);
+                break;
+            case EVENT_COMMAND:
+                task = new Event(tokens[2], tokens[3], tokens[4]);
+                break;
+            default:
+                return null; //TODO: throw exception
+        }
+        if (status.equals(TRUE)) {
+            task.markAsDone();
+        }
+        return task;
+    }
+
+    private static boolean readStatus(String line) {
+        String[] tokens = line.split("/");
+        String status = tokens[1].toUpperCase();
+        if (status.equals(TRUE)) {
+            return true;
+        }
+        else if (status.equals(FALSE)) {
+            return false;
+        }
+        return false; //TODO: throw exception
+    }
+
+    private static void writeFile() {
+        try {
+            ArrayList<String> lines = new ArrayList<String>();
+            for (Task task : tasks) {
+                lines.add(task.toString());
+            }
+            FileHandler.writeFile("data/nyan.txt", "temp/temp.txt", lines);
+        } catch (IOException e) {
+            System.out.println("IO exception");
+        }
+    }
+
+    public static void endNyan() {
+        writeFile();
+        Printer.printBye();
+    }
+
+     public static void main(String[] args) {
```

Comment:
good use of SLAP

https://nus-cs2113-ay2324s2.github.io/website/se-book-adapted/chapters/codeQuality.html#slap-hard